File: C++/프로그래머스/1/42862. 체육복/체육복.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    vector<int> trainingSet(n + 2, 1);
    
    for(auto& l : lost) trainingSet[l]--;
    for(auto& r : reserve) trainingSet[r]++;
    
    int answer = 0;
    for(int i = 1; i <= n; i++) {
        if(trainingSet[i] == 0) { //체육복이 없을 때 빌리기
            if(trainingSet[i - 1] == 2) {
                trainingSet[i - 1]--;
                trainingSet[i]++;
            } else if (trainingSet[i + 1] == 2) {
                trainingSet[i + 1]--;
                trainingSet[i]++;
            }
        }
        answer += min(trainingSet[i], 1);
    }
    return answer;
}

File: Python3/프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    lost, reserve = [l for l in lost if l not in reserve], [r for r in reserve if r not in lost]
    borrowed = [False if i in reserve else True for i in range(n + 2)]
    answer = n - len(lost)
    
    for l in sorted(lost):
        if not borrowed[l - 1]:
            borrowed[l - 1] = True
            answer += 1
        elif not borrowed[l + 1]:
            borrowed[l + 1] = True
            answer += 1
    
    return answer